### PR TITLE
#11422: Replace tt_lib in models/experimental/trocr,vgg,vit

### PR DESCRIPTION
**Pull request by mouliraj-mcw**
Issue : #11422 Tracking Issue : #11240 , Master Issue : #10532 

### Work Done:

- Replace tt_lib usage in `models/experimental/trocr`
- Replace tt_lib usage in `models/experimental/vgg`
- Replace tt_lib usage in `models/experimental/vit`

### Checklist
- [x] All Post commit CI 
- [ ]  Nightly Fast Dispatch Test
- [ ] Model regression CI testing passes

### CODE REVIEW

File: models/experimental/vgg/vgg_utils.py [hunk 1]
```diff
@@ -46,45 +46,45 @@ def cache_weights_in_weka(device, model_location_generator):
...
+            value = ttnn.Tensor(
                 value.reshape(-1).tolist(),
                 value.shape,
-                tt_lib.tensor.DataType.BFLOAT16,
-                tt_lib.tensor.Layout.ROW_MAJOR,
+                ttnn.bfloat16,
+                ttnn.ROW_MAJOR_LAYOUT,
             )
-        tt_lib.tensor.dump_tensor(file_name + str(key) + ".bin", value)
+        ttnn.experimental.tensor.dump_tensor(file_name + str(key) + ".bin", value)
```
Comment (mcw-anasuya) [inline models/experimental/vgg/vgg_utils.py:87]: Try changing ttnn.experimental.tensor.dump_tensor to ttnn.dump_tensor

File: models/experimental/vgg/vgg_utils.py [hunk 2]
```diff
@@ -104,21 +104,21 @@ def store_weights(model_version, file_name, dtype, base_addresses):
...
+            value = ttnn.Tensor(
                 value.reshape(-1).tolist(),
                 value.shape,
                 dtype,
-                tt_lib.tensor.Layout.ROW_MAJOR,
+                ttnn.ROW_MAJOR_LAYOUT,
             )
 
-        tt_lib.tensor.dump_tensor(file_name + str(key) + str(dtype) + ".bin", value)
+        ttnn.experimental.tensor.dump_tensor(file_name + str(key) + str(dtype) + ".bin", value)
```
Comment (mcw-anasuya) [inline models/experimental/vgg/vgg_utils.py:121]: Same here

**Review by KalaivaniMCW — APPROVED** [review]
Approved. Pls run nightly FD and check local run

File: models/experimental/trocr/tt/trocr_for_causal_llm.py [hunk 3]
```diff
@@ -56,33 +54,25 @@ def __init__(
...
         output_attentions: Optional[bool] = None,
         output_hidden_states: Optional[bool] = None,
         return_dict: Optional[bool] = None,
     ) -> Union[Tuple, TtCausalLMOutputWithCrossAttentions]:
-        output_attentions = (
-            output_attentions
-            if output_attentions is not None
-            else self.config.output_attentions
-        )
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
```
Comment (mcw-anasuya) [inline models/experimental/trocr/tt/trocr_for_causal_llm.py:71]: Missed parentheses, check main

File: models/experimental/trocr/tests/test_tt_trocr_learned_positional_embeddings.py [hunk 4]
```diff
@@ -65,6 +62,4 @@ def test_trocr_attention_inference(device, pcc, reset_seeds):
         else:
             logger.warning("TrOCRLearnedPositionalEmbeddings Failed!")
 
-        assert (
-            passing
-        ), f"TrOCRLearnedPositionalEmbeddings output does not meet PCC requirement {pcc}."
+        assert passing, f"TrOCRLearnedPositionalEmbeddings output does not meet PCC requirement {pcc}."
```
Comment (mcw-anasuya) [inline models/experimental/trocr/tests/test_tt_trocr_learned_positional_embeddings.py:65]: Missed parentheses, check main

File: models/experimental/vit/tt/modeling_vit.py [hunk 5]
```diff
@@ -142,11 +136,11 @@ def forward(
...
         attention_scores = ttnn.matmul(query_layer, key_layer_T, memory_config=self.out_mem_config_l1)
 
-        attention_scores = tt_lib.tensor.bcast(
+        attention_scores = ttnn.experimental.tensor.bcast(
             attention_scores,
             self.recip_sqrt_attention_head_size_tensor,
-            tt_lib.tensor.BcastOpMath.MUL,
-            tt_lib.tensor.BcastOpDim.HW,
+            ttnn.experimental.tensor.BcastOpMath.MUL,
+            ttnn.experimental.tensor.BcastOpMath.HW,
```
Comment (mcw-anasuya) [inline models/experimental/vit/tt/modeling_vit.py:143]: This should be ttnn.experimental.tensor.BcastOpDim.HW

**Review by mcw-anasuya — APPROVED** [review]

**Review by eyonland — APPROVED** [review]

**Pull request merged**